tests/marker_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests_test
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubevirt/bridge-marker/tests"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("bridge-marker", func() {
+	Describe("bridge resource reporting", func() {
+		It("should be reported only when available on node", func() {
+
+			out, err := tests.RunOnNode("node01", "sudo ip link add br-test type bridge")
```

Comment:
what if the bridge exists? will this command fail then? would it make sense to use a unique name per run to avoid potential clashes / failures because old bridge is left?

Reply:
Generating a unique bridge name.

Reply:
I think that tests should not consume, in general, system objects in an unknown state. I prefer a loud failure, or an explicit pre-cleanup function over randomness.

Reply:
> I think that tests should not consume, in general, system objects in an unknown state. I prefer a loud failure, or an explicit pre-cleanup function over randomness.

This PR is already closed.
Sebastian is changing this back in: https://github.com/kubevirt/bridge-marker/pull/8